app/components/FilterPanel.tsx

```diff
@@ -0,0 +1,204 @@
+                        ))}
+                    </select>
+                </div>
+
+                <div>
+                    <label className='block text-xs text-gray-600'>
+                        Opublikowane
+                    </label>
+                    <select
+                        value={filters.daysAgo ?? 'all'}
+                        onChange={(e) =>
+                            setFilters({
+                                ...filters,
+                                daysAgo:
+                                    e.target.value === 'all'
+                                        ? 'all'
+                                        : Number(e.target.value),
+                            })
+                        }
+                        className='mt-1 w-full rounded border p-2 text-sm'
+                    >
+                        <option value='all'>Wszystkie</option>
+                        <option value={7}>Ostatnie 7 dni</option>
+                        <option value={30}>Ostatnie 30 dni</option>
+                        <option value={365}>Ostatni rok</option>
+                    </select>
+                </div>
+
+                <div className='sm:col-span-2'>
+                    <label className='block text-xs text-gray-600'>
+                        Cena (min / max)
+                    </label>
+                    <div className='mt-1 flex gap-2'>
+                        <input
+                            type='number'
+                            placeholder='min'
+                            value={filters.minPrice ?? ''}
+                            onChange={(e) =>
+                                setFilters({
+                                    ...filters,
+                                    minPrice: e.target.value
+                                        ? Number(e.target.value)
+                                        : undefined,
+                                })
+                            }
+                            className='w-1/2 rounded border p-2 text-sm'
+                        />
+                        <input
+                            type='number'
+                            placeholder='max'
+                            value={filters.maxPrice ?? ''}
+                            onChange={(e) =>
+                                setFilters({
+                                    ...filters,
+                                    maxPrice: e.target.value
+                                        ? Number(e.target.value)
+                                        : undefined,
+                                })
+                            }
+                            className='w-1/2 rounded border p-2 text-sm'
+                        />
+                    </div>
+                </div>
+            </div>
+            <div className='mt-3 flex gap-2'>
+                <button
+                    type='button'
+                    onClick={() =>
```

Comment:
create separate clear function
```suggestion
                    onClick={clearFilters}
```